clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheStateServiceImpl.java

```diff
@@ -289,6 +430,7 @@ private void releasePool(String poolType, String poolName, ResourcePageSource re
     ServerSideConfiguration.Pool pool = resourcePageSource.getPool();
     OffHeapResource source = services.getService(OffHeapResourceIdentifier.identifier(pool.getServerResource()));
     if (source != null) {
+      unRegisterPoolStatistics(resourcePageSource);
```

Comment:
Scratch what I said ... again 😉 
Actually, this will be called when the entity is destroyed for shared pools, but not for dedicated pools AFAICT.
However, it will be properly called if a `ServerStore` using a dedicated pool is destroyed explicitly.
So there is still one path to cover - linked to my comment on `unregisterStoreStatistics`.